c7n/resources/apigw.py

```diff
@@ -487,8 +488,150 @@ def augment(self, related, resource):
             }
 
 
+@RestStage.filter_registry.register('waf-enabled')
+class WafEnabled(Filter):
+    """Filter API Gateway stage by waf-regional web-acl
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: filter-apigw-waf-regional
+                resource: rest-stage
+                filters:
+                  - type: waf-enabled
+                    state: false
+                    web-acl: test
+    """
+    schema = type_schema(
+        'waf-enabled', **{
+            'web-acl': {'type': 'string'},
+            'state': {'type': 'boolean'}})
+
+    permissions = ('waf:ListWebACLs', 'waf:GetWebACL')
+
+    def process(self, resources, event=None):
+        target_acl = self.data.get('web-acl')
+        state = self.data.get('state', False)
+
+        results = []
+        wafs = self.manager.get_resource_manager('waf-regional').resources(augment=False)
```

Comment:
in this context I think you want the default augment which will do the get_web_acl fetch for you, instead of having to reimplement that loop here.

Reply:
it does look like the augment methods on these resource managers aren't correct as they need to do a super().augment(resources) so that the resource detail spec is invoked and the webacl arn is present.

Reply:
Thanks for pointing this. I have fixed this.

Reply:
@kapilt Can you review this code change?

Reply:
thanks for the updates, looks good.